### PR TITLE
[Java] Support registration in thread safe fury

### LINKED ISSUE #1279: [Java] support class registration in ThreadSafeFury
## Is your feature request related to a problem? Please describe.
`Fury` provide a `register` API to register class and serializer, but `ThreadSafeFury` doesn't provide such API.

Although we provide such doc for registration:
```java
ThreadSafeFury fury =
    new ThreadLocalFury(
        classLoader -> {
          Fury f = Fury.builder().requireClassRegistration(true).build();
          // register and generate serializer code.
          f.register(Foo.class, true);
          return f;
        });
System.out.println("Init fury at build time");
```
Users still missed this doc often and ask for help how to register class using `ThreadSafeFury`. Also it's confusing that `Fury` and `ThreadSafeFury` doesn't provide consistent API.

## Describe the solution you'd like
Provide `register` API for `ThreadSafeFury`

## Additional context
#704

### PR DESCRIPTION
**Pull request by chaokunyang**
Closes #1279

### CODE REVIEW

File: java/fury-core/src/main/java/org/apache/fury/AbstractThreadSafeFury.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package org.apache.fury;
```
Comment (tisonkun) [inline java/fury-core/src/main/java/org/apache/fury/AbstractThreadSafeFury.java:1]: Please properly add license header or include necessary info if it's copied.

**Review by tisonkun — COMMENTED** [review]
LGTM for the technical part,

File: java/fury-core/src/main/java/org/apache/fury/AbstractThreadSafeFury.java (hunk 1 above)
Comment (chaokunyang) [reply java/fury-core/src/main/java/org/apache/fury/AbstractThreadSafeFury.java:20]: header added.  Thanks for this catch, I always forget add header for new file.

Comment (tisonkun) [reply java/fury-core/src/main/java/org/apache/fury/AbstractThreadSafeFury.java:20]: I don't look into details but CI should fail if license headers missing.

**Review by tisonkun — APPROVED** [review]

File: java/fury-core/src/main/java/org/apache/fury/AbstractThreadSafeFury.java (hunk 1 above)
Comment (pjfanning) [reply java/fury-core/src/main/java/org/apache/fury/AbstractThreadSafeFury.java:20]: * the code style check was failing until the header was added
* the Graal VM CI is failing still

**Review by pjfanning — CHANGES_REQUESTED** [review]
Graal VM check is failing

File: java/fury-core/src/main/java/org/apache/fury/AbstractThreadSafeFury.java (hunk 1 above)
Comment (chaokunyang) [reply java/fury-core/src/main/java/org/apache/fury/AbstractThreadSafeFury.java:20]: CI succeed now

**Comment by chaokunyang** [comment]
CI has been fixed

**Review by pjfanning — APPROVED** [review]
lgtm

**Pull request merged**